### DISCUSSION TITLE
Where is the IP?

### DISCUSSION THREAD

**Discussion by Karem-sobhy** [shlinkio/discussions#10] [Help wanted] [answered]
just wanted to know where to find the IPs of the visits because I can't find them.

**Comment by acelaya** [comment]
The IP address is in the `visits` table, in the `remote_addr` column.

**Reply by Karem-sobhy** [reply]
Oh! So it's not showing in the UI?

**Reply by acelaya** [reply]
My mistake. I didn't notice you were asking this in the web client repo.

I'm afraid not. Shlink does not expose the IP addresses for security and privacy regulations reasons.

They are only used for geolocation, and then, they are anonymized and stored just in case the location needs to be resolved again.

**Reply by eiqnepm** [reply]
> My mistake. I didn't notice you were asking this in the web client repo.
> 
> I'm afraid not. Shlink does not expose the IP addresses for security and privacy regulations reasons.
> 
> They are only used for geolocation, and then, they are anonymized and stored just in case the location needs to be resolved again.

Would you be against adding an environment variable for custom hosted `shlink-web-client` instances to explicitly enable an IP address column in the visits list? I see that `shlink` has the environment variable `ANONYMIZE_REMOTE_ADDR`, which if I assume correctly, when set to `false` stores the visitors IP address in plaintext, however `shlink-web-client` has no way to display this information.

I'm currently messing around with a few link shortener projects, `shlink` does seem to come out on top in most regards, however I'd still like to be able to view visit IP addresses from `shlink-web-client`.

Nevertheless, great project, thanks a bunch for your work and for making it FOSS!

**Reply by acelaya** [reply]
As explained above, Shlink does not expose the IP addresses at all, for privacy reasons, so it's trickier to change.

What's exactly the use case for this? Can you explain how would knowing the actual IP address help?

**Reply by gcarrarom** [reply]
It would be interesting to see this for correlating "sessions" between links. I've seen this use-case when I have an user come from one url and I'm not sure if that's the same going to the other. Maybe not the IP address, but could a "session_id" be created?

**Reply by acelaya** [reply, marked as answer]
@gcarrarom this was discussed some time ago in some issue that I'm trying to find.

As a result, I created this https://github.com/shlinkio/shlink/issues/725, which was never implemented due to other priorities and luck of traction.

If anything was implemented, I would be inclined to go with option 2, and expose a hash identifying the visitor, but without exposing its IP address.

**Reply by theofficialgman** [reply]
> The IP address is in the `visits` table, in the `remote_addr` column.

Where can that information be obtained? I see no such column in the cli short-url:visits or REST api. I would like to see the anonymized (or real if that option is enabled) addresses.
